test/steps_test.go

```diff
@@ -1555,37 +1422,54 @@ func checkExpectedVsActualAssetParams() error {
 		return fmt.Errorf("expected default frozen state %v but actual default frozen state was %v",
 			expectedParams.DefaultFrozen, actualParams.DefaultFrozen)
 	}
-	managerMatch := expectedParams.ManagerAddr == actualParams.ManagerAddr
+	managerMatch := expectedParams.Manager == actualParams.Manager
 	if !managerMatch {
 		return fmt.Errorf("expected asset manager was %v but actual asset manager was %v",
-			expectedParams.ManagerAddr, actualParams.ManagerAddr)
+			expectedParams.Manager, actualParams.Manager)
 	}
-	reserveMatch := expectedParams.ReserveAddr == actualParams.ReserveAddr
+	reserveMatch := expectedParams.Reserve == actualParams.Reserve
 	if !reserveMatch {
 		return fmt.Errorf("expected asset reserve was %v but actual asset reserve was %v",
-			expectedParams.ReserveAddr, actualParams.ReserveAddr)
+			expectedParams.Reserve, actualParams.Reserve)
 	}
-	freezeMatch := expectedParams.FreezeAddr == actualParams.FreezeAddr
+	freezeMatch := expectedParams.Freeze == actualParams.Freeze
 	if !freezeMatch {
 		return fmt.Errorf("expected freeze manager was %v but actual freeze manager was %v",
-			expectedParams.FreezeAddr, actualParams.FreezeAddr)
+			expectedParams.Freeze, actualParams.Freeze)
 	}
-	clawbackMatch := expectedParams.ClawbackAddr == actualParams.ClawbackAddr
+	clawbackMatch := expectedParams.Clawback == actualParams.Clawback
 	if !clawbackMatch {
 		return fmt.Errorf("expected revocation (clawback) manager was %v but actual revocation manager was %v",
-			expectedParams.ClawbackAddr, actualParams.ClawbackAddr)
+			expectedParams.Clawback, actualParams.Clawback)
 	}
 	return nil
 }
 
+func findAssetID(assets []modelsV2.AssetHolding, assetID uint64) (foundAsset modelsV2.AssetHolding, err error) {
+	found := false
+	for _, asset := range assets {
+		idx := asset.AssetId
+		if idx == assetID {
+			found = true
+			foundAsset = asset
+			break
+		}
+	}
+	if !found {
+		return modelsV2.AssetHolding{}, fmt.Errorf("asset ID %d was not found", assetID)
+	}
+	return foundAsset, nil
```

Comment:
nit: simplification, 

```suggestion
	for _, asset := range assets {
		if asset.AssetId == assetID {
			return asset, nil
		}
	}
	return modelsV2.AssetHolding{}, fmt.Errorf("asset ID %d was not found", assetID)
```

Reply:
Nice catch, thanks!